File: lib/models/sponsor.dart
class Sponsor {
  final int id;
  final String nameEn;
  final String nameAr;
  final String? detailsEn;
  final String? detailsAr;
  final String type;
  final String photoPath;

  const Sponsor({
    required this.id,
    required this.nameEn,
    required this.nameAr,
    required this.detailsEn,
    required this.detailsAr,
    required this.type,
    required this.photoPath,
  });

  factory Sponsor.fromJson(Map<String, dynamic> json) {
    return Sponsor(
      id: json['id'] as int,
      nameEn: (json['name_en'] ?? '').toString(),
      nameAr: (json['name_ar'] ?? '').toString(),
      detailsEn: (json['details_en']?.toString().isEmpty ?? true)
          ? null
          : json['details_en'].toString(),
      detailsAr: (json['details_ar']?.toString().isEmpty ?? true)
          ? null
          : json['details_ar'].toString(),
      type: (json['type'] ?? '').toString(),
      photoPath: (json['photo_path'] ?? '').toString(),
    );
  }
}

File: project_backend_part/app/Models/Sponsor.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Sponsor extends Model
{
    use HasFactory;

    public const TYPE_STRATEGIC_PARTNER = 'Strategic partner';
    public const TYPE_GOLD_PARTNER = 'Gold Partner';
    public const TYPE_DIAMOND_PARTNER = 'Diamond Partner';

    /**
     * @var array<int, string>
     */
    public const TYPES = [
        self::TYPE_STRATEGIC_PARTNER,
        self::TYPE_GOLD_PARTNER,
        self::TYPE_DIAMOND_PARTNER,
    ];

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'name_en',
        'name_ar',
        'details_en',
        'details_ar',
        'type',
        'photo_path',
    ];
}
